src/Playroom/Playroom.css.ts

```diff
@@ -27,71 +34,58 @@ globalStyle('body', {
   margin: 0,
 });
 
-export const root = sprinkles({
-  height: 'viewport',
-  width: 'viewport',
-});
-
-export const previewContainer = sprinkles({
-  position: 'absolute',
-  inset: 0,
-});
-
-export const editorSize = createVar();
-
-export const previewContainerPosition = styleVariants({
+export const root = styleVariants({
   right: {
-    right: `max(${editorSize}, ${MIN_WIDTH}px)`,
+    flexDirection: 'row',
   },
   bottom: {
-    bottom: `max(${editorSize}, ${MIN_HEIGHT}px)`,
+    flexDirection: 'column',
   },
   undocked: {},
 });
 
-export const resizableContainer = style([
+// Prevents the editor growing off screen
+// when resizable value increases past maximum size
+export const previewContainer = style({
+  minWidth: 0,
+  minHeight: 0,
+});
+
+export const editorSize = createVar();
+
+export const resizable = style([
   sprinkles({
-    bottom: 0,
-    right: 0,
     overflow: 'hidden',
     boxShadow: 'small',
-    transition: 'slow',
   }),
-  // @ts-expect-error Shouldnt need to but types do not like `!important`
-  {
-    position: 'absolute !important', // override re-resizable's inline style
-  },
 ]);
 
-export const resizableContainer_isHidden = style({});
-
-export const resizableContainer_isRight = style([
-  sprinkles({
-    top: 0,
-  }),
-  {
-    maxWidth: '90vw',
-    selectors: {
-      [`&${resizableContainer_isHidden}`]: {
-        transform: 'translateX(100%)',
-      },
-    },
+// override re-resizable's inline style
+// clamp ensures editor transitioning to correct size
+// before applying minimum and maximum sizes
+export const resizableSize = styleVariants({
+  right: {
+    width: `clamp(${MIN_WIDTH}px, ${editorSize}, ${MAX_WIDTH}) !important`,
   },
-]);
+  bottom: {
+    height: `clamp(${MIN_HEIGHT}px, ${editorSize}, ${MAX_HEIGHT}}) !important`,
```

Comment:
This rule is broken with the extra brace in it, but wondering whether we need these clamps.

Can we add a `minWidth` to the `resizeableAvailable` rule below and then have these just set their value as `editorSize`?

```suggestion
    height: `clamp(${MIN_HEIGHT}px, ${editorSize}, ${MAX_HEIGHT}) !important`,
```

Reply:
In fact, playing with it now I dont think we need this rule at all anymore.

Reply:
The issue should be apparent when size percentage is less than the minimum size. You can see it if you set the editor size to the minimum and then shrink the viewport size to be very small